modules/dreamview/frontend/src/renderer/tileground.js

```diff
@@ -82,28 +77,15 @@ export default class TileGround {
             if (i !== row || j !== col) {
                 return;
             }
-            const itemPos = coordinates.applyOffset({
-                x: item.x,
-                y: item.y,
-                z: 0,
-            });
-            loadObject(material, object, scales,
-                mesh => {
-                    mesh.rotation.x = Math.PI / 2;
-                    mesh.rotation.y = item.heading;
-                    mesh.position.set(itemPos.x, itemPos.y, itemPos.z);
-                    if (item.id) {
-                        // store signal id as name in mesh
-                        mesh.name = item.id;
-                    }
-                    mesh.matrixAutoUpdate = false;
-                    mesh.updateMatrix();
+            item.type = type;
+            newItems.push(item);
 
-                    // a tile may have multiple signal or stopSign
-                    currentItems[`${key}_${item.x}_${item.y}`] = mesh;
-                    scene.add(mesh);
-                });
+            if (!this.currentItems[type]) {
+                this.currentItems[type] = [];
+            }
+            this.currentItems[type].push(`${key}_${item.id}`);
```

Comment:
Seems better to push an object with fields like key & id as opposed doing string concat and split.